docs/permission/writing-a-policy.md

```diff
@@ -3,3 +3,137 @@ id: writing-a-policy
 title: Writing a permission policy
 description: How to write your own permission policy as a Backstage integrator
 ---
+
+In the [previous section](./getting-started.md), we were able to set up the permission framework and make a simple change to our `TestPermissionPolicy` to confirm that policy is indeed wired up correctly.
+
+That policy looked like this:
+
+```typescript
+// packages/backend/src/plugins/permission.ts
+
+class TestPermissionPolicy implements PermissionPolicy {
+  async handle(request: PolicyAuthorizeQuery): Promise<PolicyDecision> {
+    if (request.permission.name === 'catalog.entity.delete') {
+      return {
+        result: AuthorizeResult.DENY,
+      };
+    }
+
+    return { result: AuthorizeResult.ALLOW };
+  }
+}
+```
+
+## What's in a policy?
+
+Let's break this down a bit further. The request object of type [PolicyAuthorizeQuery](https://backstage.io/docs/reference/plugin-permission-node.policyauthorizequery) is a simple wrapper around [the Permission object](https://backstage.io/docs/reference/plugin-permission-common.permission). This permission object encapsulates information about the action that the user is attemping to perform (See [the Concepts page](./concepts.md) for more details).
+
+In the policy above, we are checking to see if the provided action is a catalog entity delete action, which is the permission that the catalog plugin authors have created to represent the action of unregistering a catalog entity. If this is the case, we return a [Definitive Policy Decision](https://backstage.io/docs/reference/plugin-permission-node.definitivepolicydecision) of DENY. In all other cases, we return ALLOW (resulting in an allow-by-default behavior).
+
+As we confirmed in the previous section, we know that this now prevents us from unregistering catalog components. Hooray! But you may notice that this prevents _anyone_ from unregistering a component, which is not a very realistic policy. Let's improve this policy by disabling the unregister action _unless you are the owner of this component_.
+
+## Conditional decisions
+
+Let's change the policy to the following:
+
+```diff
+- import { IdentityClient } from '@backstage/plugin-auth-node';
++ import { BackstageIdentityResponse, IdentityClient } from '@backstage/plugin-auth-node';
++ import { catalogConditions, createCatalogPolicyDecision } from '@backstage/plugin-catalog-backend';
+
+  ...
+
+  class TestPermissionPolicy implements PermissionPolicy {
+-   async handle(request: PolicyAuthorizeQuery): Promise<PolicyDecision> {
++   async handle(
++     request: PolicyAuthorizeQuery,
++     user?: BackstageIdentityResponse,
++    ): Promise<PolicyDecision> {
+      if (request.permission.name === 'catalog.entity.delete') {
+-       return {
+-         result: AuthorizeResult.DENY,
+-       };
++       return createCatalogPolicyDecision(
```

Comment:
I wonder if we should mention the "raw" way of returning a conditional decision without using any helper first and then add the helper in a second step?

Reply:
The "raw" condition objects feel internal IMO. It didn't seem valuable to dig into those during the demo since we'd rather people use the type safe factory functions. But if there's a good reason to showcase this then I'm sure we can work it in!

Reply:
then we should also align the docs for plugin authors, since there we are not using any utility function.
I will open an issue for that!